packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -113,6 +117,97 @@ export default class FulfillmentModuleService<
     return joinerConfig
   }
 
+  protected static normalizeFiltersAndConfigForShippingOptionsFiltering(
+    filters: FilterableShippingOptionProps = {},
+    config: FindConfig<ShippingOptionDTO> = {}
+  ) {
+    let { fulfillment_set_id, fulfillment_set_type, context, ...where } =
+      filters
+
+    let normalizedFilters = { ...where } as FilterQuery
+
+    const fulfillmentSetConstraints = {}
+
+    if (fulfillment_set_id || fulfillment_set_type) {
+      if (fulfillment_set_id) {
+        fulfillmentSetConstraints["id"] = fulfillment_set_id
+      }
+
+      if (fulfillment_set_type) {
+        fulfillmentSetConstraints["type"] = fulfillment_set_type
+      }
+
+      normalizedFilters = {
+        ...normalizedFilters,
+        service_zone: {
+          fulfillment_set: fulfillmentSetConstraints,
+        },
+      }
+    }
+
+    const normalizedConfig = { ...config }
+    normalizedConfig.relations = [
+      "rules",
+      "type",
+      "shipping_profile",
+      "service_provider",
+      "service_zone.fulfillment_set",
+      ...(normalizedConfig.relations ?? []),
+    ]
+    // The assumption is that there won't be an infinite amount of shipping options. So if a context filtering needs to be applied we can retrieve them all.
+    normalizedConfig.take =
+      normalizedConfig.take ?? (context ? null : undefined)
+
+    return {
+      filters: normalizedFilters,
+      config: normalizedConfig,
+      context,
+    }
+  }
+
+  @InjectManager("baseRepository_")
+  // @ts-ignore
+  async listShippingOptions(
+    filters: FilterableShippingOptionProps = {},
+    config: FindConfig<ShippingOptionDTO> = {},
+    sharedContext?: Context
+  ): Promise<FulfillmentTypes.ShippingOptionDTO[]> {
+    const {
+      filters: normalizedFilters,
+      config: normalizedConfig,
+      context,
+    } = FulfillmentModuleService.normalizeFiltersAndConfigForShippingOptionsFiltering(
+      filters,
+      config
+    )
+
+    let shippingOptions = await this.shippingOptionService_.list(
+      normalizedFilters,
+      normalizedConfig,
+      sharedContext
+    )
+
+    // Apply rules context filtering
+    if (context) {
+      shippingOptions = shippingOptions.filter((shippingOption) => {
+        if (!shippingOption.rules?.length) {
+          return true
+        }
+
+        return isContextValid(
+          context,
+          shippingOption.rules.map((r) => r)
```

Comment:
```suggestion
          shippingOption.rules
```
this should suffice right?

Reply:
It is a collection under the hood, I just do that yo extract them, the other options would be to force cast them ahah

Reply:
ah, i see!